app/uk/gov/hmrc/transitmovementsvalidator/controllers/MessagesController.scala

```diff
@@ -66,8 +78,29 @@ class MessagesController @Inject() (cc: ControllerComponents, xmlValidationServi
     contentTypeRoute {
       case Some(MimeTypes.XML)  => validateMessage(messageType, xmlValidationService)
       case Some(MimeTypes.JSON) => validateMessage(messageType, jsonValidationService)
+      case None                 => validateObjectStoreMessage(messageType, xmlValidationService)
     }
 
+  private val failCase: PresentationError => Result = presentationError =>
+    Status(presentationError.code.statusCode)(Json.toJson(presentationError)(PresentationError.presentationErrorWrites))
+
+  def validateObjectStoreMessage(messageType: String, validationService: ValidationService): Action[AnyContent] = Action.async {
+    implicit request =>
+      implicit val hc = HeaderCarrierConverter.fromRequest(request)
+      (for {
+        uri      <- getObjectStoreUri
+        contents <- objectStoreService.getContents(uri)(executionContext, hc).asPresentation
```

Comment:
(executionContext, hc) - these are implicit so you don't need to pass them to the request but equally there's nothing wrong with passing them

Reply:
Agree - will remove.